core/src/main/scala/Key.scala

```diff
@@ -8,7 +8,12 @@ class Key(val underlying: Array[Byte]) extends AnyVal with Ordered[Key] {
   def is(that: Key): Boolean = underlying sameElements that.underlying
 
   override def compare(that: Key): Int = {
-    // TODO: Implement compare
-    underlying.length
+    val result = underlying
+      .zip(that.underlying)
+      .map { case (a, b) =>
+        a - b
+      }
+      .find(_ != 0)
+    result.getOrElse(0)
   }
```

Comment:
Unneccesary code block and the variable.

```scala
  override def compare(that: Key): Int = underlying
    .zip(that.underlying)
    .map { case (a, b) =>
      a - b
    }
    .find(_ != 0)
    .getOrElse(0)
```

Reply:
sure